docs/references/expo/expo-impersonation.mdx

```diff
@@ -0,0 +1,509 @@
+  const { isSignedIn } = useAuth();
+
+  if (!isSignedIn) {
+    return <Redirect href={"/dashboard"} />;
+  }
+
+  return <Stack />;
+}
+```
+
+### Create a `(dashboard)` page
+
+Create the main page of the dashboard, which will hold most of your impersonation code. This example contains the skeleton of the final result, which will be fleshed out with each step in this guide.
+
+> To skip this process, see [the full example](#full-example) at the end of this guide.
+
+```tsx {{ filename: "app/(dashboard)/index.tsx" }}
+import React from "react";
+import { Button, StyleSheet, Text, TouchableOpacity, View } from "react-native";
+import { Link, useRouter } from "expo-router";
+import { useAuth, useUser, useSignIn, useSessionList } from "@clerk/clerk-expo";
+import { UserDataJSON } from "@clerk/types";
+import * as Linking from "expo-linking";
+import useImpersonation from "@/hooks/useImpersonation";
+
+// this hook will return all of the user data for our impersonator, using an API route
+function useImpersonatedUser() {}
+
+// this hook will generate the token needed to create an impersonated session, also using an API route
+function userImpersonation() {}
+
+export default function Dashboard() {
+	// this state will allow us to display some information about the current impersonator
+	const [impersonator, setImpersonator] = React.useState<UserDataJSON | string>("");
+	// our custom Clerk hooks provide us with variables and functions to handle our authentication flows
+  const { signOut, actor } = useAuth();
+  const { isLoaded, signIn, setActive } = useSignIn();
+  const { user } = useUser();
+  const { sessions } = useSessionList();
+
+  // using expo's router will allow us to navigate properly once we've newly authenticated
+  const router = useRouter();
+
+  const actorRes = useImpersonation(actor?.sub || undefined, user?.id);
+  const actorUserData = useImpersonatedUser(actor?.sub || "", setImpersonator);
+
+  function extractTicketValue(input: string): string | undefined {}
+
+	async function onSignoutPress(sessionId: string) => {}
+
+	return (
+		<View>
+      <Link href="/account">
+        <Text>Account</Text>
+      </Link>
+      <Text>Hello {user?.firstName}</Text>
+
+      {sessions?.map((sesh) => (
+        <TouchableOpacity
+          onPress={() => onSignOutPress(sesh.id)}
+          key={sesh.id}
+        >
+          <Text>
+            Sign out of {sesh?.user?.primaryEmailAddress?.emailAddress}
+          </Text>
+        </TouchableOpacity>
+      ))}
+
+      {actorRes && (
+        <Button
+          title="Impersonate"
+          onPress={async () => await impersonateUser()}
+        />
+      )}
+    </View>
+	)
+}
+```
+
+### Create a hook to generate an actor token
+
+Create a custom hook called `useImpersonation()` that will call an API route to generate and return an actor token from Clerk.
+
+```tsx {{ filename: "app/(dashboard)/index.tsx" }}
+
+import { useState, useEffect } from "react";
+
+export type Actor = {
```

Comment:
Can the type be trimmed to just the pieces needed?

Reply:
I wanted to doc the whole type so folks could expand on it and use what properties they need, since I didn't actual build anything _with_ the impersonators data. You feel strongly to knock it down to only the used props?

Reply:
There is no right or wrong, more just thinking of keeping the examples lean. Maybe trim the type in the code and and have a separate section that has the whole type for those who want it? Keeping the whole type also works -- maybe add a note with your intent to it if you do that. Like 'Below is the whole Actor type, but on fields X and Y are required. The full type exists for reference in case needed.'

Reply:
I think that's a great compromise, I'll do that 🫡